src/AdvancementAPI.java

```diff
@@ -89,6 +95,9 @@ public AdvancementAPI icon(String icon) {
         this.icon = icon;
         return this;
     }
+    public AdvancementAPI icon(Material icon){
+        return this.icon("minecraft:"+icon.name().toLowerCase());
```

Comment:
This isn't always guaranteed to function because the Material enum constants are not consistent with Mojang's item ID's. It would be better to use a NamespacedKey here to allow for things such as `NamespacedKey.minecraft("apple")`, for example